### PR TITLE
Updating the 'lessons learned' paper

### PR DESCRIPTION
**Pull request by gvwilson**
Work in progress: feedback welcome.

### CODE REVIEW

File: bib/software-carpentry-lessons-learned/software-carpentry-lessons-learned.tex [hunk 1]
```diff
@@ -231,76 +214,158 @@ \subsection*{Version 4: orange light}
...
+  designed with this in mind.  In particular, by the time there 400
+  people are on the books, most will be dipping in and out of
+  conversations rather than taking part on a daily or weekly basis, so
+  frameworks and procedures must become simple and stable.
+
+\item
+  Every partner organization has different needs and
+  constraints\footnote{We have learned much more than we ever wanted
+    to about accounting rules at various universities\ldots}.
+  ``Standard'' partnership agreements is essential therefore have to
```
Comment (jduckles) [inline bib/software-carpentry-lessons-learned/software-carpentry-lessons-learned.tex]: partnership agreements `s/is/are` essential.

**Comment by jduckles** [comment]
One more thing, in section one you claim, via a a quote, "130% improvement in test scores". Then in S2.7 you say "we do not know how to measure the impact of our workshops". You measured something in S1!  Perhaps a reword in S2.7 to "we struggle to effectively measure the short-term and long-term impacts of our workshops on learners"

File: bib/software-carpentry-lessons-learned/software-carpentry-lessons-learned.tex [hunk 2]
```diff
@@ -423,60 +495,100 @@ \section*{What we do}
...
-Learners tell us that it is important to them to leave the workshop
-with their own working environment set up. We therefore continue to
-teach on all three major platforms (Linux, Mac OS X, and Windows),
-even though it would be simpler to require learners to use just
-one. We have experimented with virtual machines on learners' computers
-to reduce installation problems, but those introduce problems of their
-own: older or smaller machines simply aren't fast enough.  We have
-also tried using virtual machines (VMs) in the cloud, but this makes
-us dependent on university-quality WiFi.
+Some instructors VMs in the cloud.  This is an authentic way of
```
Comment (wking) [inline bib/software-carpentry-lessons-learned/software-carpentry-lessons-learned.tex]: “instructors VMs” → “instructors use VMs”.

**Comment by PBarmby** [comment]
Can't figure out how to comment on diff or deal with fork (blame administrator brain). So, typos/wording:
Line 145: suggest different wording than "fell between 2 stools" - might be baffling to non-native speakers.
Line 238: need *to* come up
Line 315: drop comma at end
Line 484: define "this"

In "why teach?" I'd suggest changing "reputation" to "network", or add in "network" since I think that's an important aspect.

In "why not?" (line 719) another possible reason is that (at least in higher ed) you don't step in the same river twice. If you're teaching a full or half-year course, you're not coming back to the same material for 6 months to a year, and the context in which it's taught may well be different. Not quite
the same situation as a workshop whose content is ~ fixed.

Don't know if this belongs in the paper, but one lesson I have learned from SWC/@gvwilson, is that managing a project like this is sometimes about *asking the right people to do things*. People may not always volunteer but they are more likely to say yes if asked directly.

**Comment by justincely** [comment]
Azalee's affiliation isn't STScI anymore - she's at UC-Davis, and I feel like Matt Davis has moved too.

**Comment by jdblischak** [comment]
Sorry for getting back to you so late. I really enjoyed reading the paper. It's nice to reflect on how much we've learned over the years.

I only have one major comment. In section 8.5 Testing, you explain the reasoning why we stopped requiring testing in the core curriculum for intermediate workshops. I think you should also include that removing testing from the core curriculum was due to the increasing number of novice workshops being taught. Teaching novices to write their first function in the morning, and then in the afternoon of that same day trying to teach them to write additional functions to test the original one was too much (at least in my experience) because novices are still struggling to understand encapsulation, default arguments, function syntax, etc.

**Pull request merged**